### ISSUE TITLE
A potential race condition in taosIpStr()

### ISSUE THREAD

**Issue by jieatcoderrect** [taosdata/TDengine#3900]
**Bug Description**
The function taosIpStr() converts a 32-bit uint32_t to an IP address string like "127.0.0.1".  It defines two static variables, ipStrArray and ipStrInex, to store the convertedIP string.

When a thread calls this function the array pointed by ipStrIndex is used to store the IP string.

The problem is that "ipStrIndex" isn't protected by any synchronization mechanism. Both threads may use the same array if they call this function simultaneously.

**To Reproduce**
For example, dnodeProcessMnodeWriteQueue thread and httpAcceptHttpConnection thread can call this funciton at the same time.

**Expected Behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/57909684/96377280-697d3f00-114a-11eb-9ebc-b7b471420115.png)

**Environment (please complete the following information):**
 - OS: Any
 - Memory, CPU, current Disk Space
 - TDengine Version develop branch and commit 95aa2442e

**Additional Context**

**Comment by sangshuduo** [comment]
Interesting, is it discovered by a static analysis software?

**Comment by jieatcoderrect** [comment]
> Interesting, is it discovered by a static analysis software?

Yes. I'm using coderrect (you can download 0.8.0 from the website https://coderrect.com/download/ or the latest version from the link https://public-installer-pkg.s3.us-east-2.amazonaws.com/coderrect-linux-develop.tar.gz for free).

BTW, there is another bug in src/sync/CMakeLists.txt. It should be

  LIST(REMOVE_ITEM SRC src/tarbitrator.c)

Otherwise, tarbitrator.c won't be removed from SRC and a "main()" will be included in libsync.a.

I found ~150 race conditions, some of them are false positives, some are trade-off between performance and safety (e.g. access to control flags), but some are real. 

I may open other tickets when going through the race list.

**Comment by jieatcoderrect** [comment]
Open #3922 to track the src/sync/CMakeLists.txt bug.

**Comment by sangshuduo** [comment]
thanks!
